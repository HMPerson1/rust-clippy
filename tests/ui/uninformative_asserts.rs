#![warn(clippy::uninformative_asserts)]
#![allow(clippy::blacklisted_name)]

fn main() {
    let foo = 0u8;
    let a = 0u8;
    let b = 0u8;
    // lint
    {
        assert!(some_condition(foo));
        assert!(some_condition(foo),);
        debug_assert!(some_condition(foo));
        assert_eq!(a, bar(b));
        assert_eq!(a, bar(b),);
        assert_ne!(a, bar(b));
        debug_assert_eq!(a, bar(b));
        debug_assert_ne!(a, bar(b));
    }

    // ok
    {
        assert!(some_condition(foo), "_");
        assert!(some_condition(foo), "{} {}", 1, 2);
        debug_assert!(some_condition(foo), "_");
        assert_eq!(a, bar(b), "_");
        assert_eq!(a, bar(b), "{} {}", 1, 2);
        assert_ne!(a, bar(b), "_");
        debug_assert_eq!(a, bar(b), "_");
        debug_assert_ne!(a, bar(b), "_");
    }
}

#[test]
fn test_something() {
    assert_eq!(bar(0), 0);
}

fn some_condition(_x: u8) -> bool {
    true
}
fn bar(x: u8) -> u8 {
    x
}
