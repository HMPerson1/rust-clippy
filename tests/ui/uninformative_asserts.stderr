error: `assert!` called without custom panic message
  --> $DIR/uninformative_asserts.rs:10:9
   |
LL |         assert!(some_condition(foo));
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::uninformative-asserts` implied by `-D warnings`
   = help: consider adding a custom panic message

error: `assert!` called without custom panic message
  --> $DIR/uninformative_asserts.rs:11:9
   |
LL |         assert!(some_condition(foo),);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider adding a custom panic message

error: `assert!` called without custom panic message
  --> $DIR/uninformative_asserts.rs:12:9
   |
LL |         debug_assert!(some_condition(foo));
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider adding a custom panic message

error: `assert!` called without custom panic message
  --> $DIR/uninformative_asserts.rs:13:9
   |
LL |         assert_eq!(a, bar(b));
   |         ^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider adding a custom panic message

error: `assert!` called without custom panic message
  --> $DIR/uninformative_asserts.rs:14:9
   |
LL |         assert_eq!(a, bar(b),);
   |         ^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider adding a custom panic message

error: `assert!` called without custom panic message
  --> $DIR/uninformative_asserts.rs:15:9
   |
LL |         assert_ne!(a, bar(b));
   |         ^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider adding a custom panic message

error: `assert!` called without custom panic message
  --> $DIR/uninformative_asserts.rs:16:9
   |
LL |         debug_assert_eq!(a, bar(b));
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider adding a custom panic message

error: `assert!` called without custom panic message
  --> $DIR/uninformative_asserts.rs:17:9
   |
LL |         debug_assert_ne!(a, bar(b));
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider adding a custom panic message

error: aborting due to 8 previous errors

